provision-osx/tasks/configure_buildfarm_node.yml

```diff
@@ -1,118 +1,106 @@
 ---
 
--
-  debug:
-    msg:
-    - Configuring macOS nodes as part of the build farm will be skipped.
-    - credential_private_key_path, credential_public_key_path and credential_passphrase must be defined.
-    - Run this job again to configure the build farm nodes.
-  when:
-  - credential_private_key_path is not defined
-  - credential_private_key_path == ''
-  - credential_public_key_path is not defined
-  - credential_public_key_path == ''
-  - credential_passphrase is not defined
-  - credential_passphrase == ''
+- include: configure_node_env_vars.yml
 
 -
   block:
   -
-    name: Get jenkins url
-    shell: "oc get route jenkins --namespace={{ project_name }} -o jsonpath='{.spec.host}'"
-    delegate_to: "{{ groups['master'][0] }}"
-    register: route_output
-    changed_when: False
-
-  - include: configure_node_env_vars.yml
+    name: Generate SSH key pair for Jenkins
+    command: ssh-keygen -t rsa -f /Users/{{ ansible_ssh_user }}/.ssh/jenkins_id_rsa -N "{{ credential_passphrase | default('') }}"
+    args:
+      creates: "/Users/{{ ansible_ssh_user }}/.ssh/jenkins_id_rsa"
 
   -
-    block:
-    -
-      name: Get the jenkins client cli jar
-      get_url:
-        url: "{{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }}/jnlpJars/jenkins-cli.jar"
-        dest: /tmp/jenkins-cli.jar
-        force: yes
-        validate_certs: no
-        mode: 0775
+    name: Output generated private key
+    command: cat /Users/{{ ansible_ssh_user }}/.ssh/jenkins_id_rsa
+    register: private_key
+    changed_when: False
 
-    -
-      name: Retrieve provided private key
-      shell: "cat {{ credential_private_key_path }}"
-      register: private_key
-      changed_when: False
+  -
+    name: Ouput generated public key
+    command: cat /Users/{{ ansible_ssh_user }}/.ssh/jenkins_id_rsa.pub
+    register: public_key
+    changed_when: False
 
-    -
-      name: Retrieve provided public key
-      shell: "cat {{ credential_public_key_path }}"
-      register: public_key
-      changed_when: False
+  -
+    name: Set credential_private_key and credential_passphrase
+    set_fact:
+      credential_private_key: "{{ private_key.stdout }}"
+      credential_public_key: "{{ public_key.stdout }}"
+  run_once: true
+  when:
+    - (credential_private_key is undefined) or (credential_public_key is undefined)
```

Comment:
will this re-generate the key every time when the task is invoked? Perhaps we should only run it if the generated key files doesn't exist?